### DISCUSSION TITLE
rest client centralize i/o exceptions handling

### DISCUSSION THREAD

**Discussion by najibk** [quarkusio/quarkus#23923] [Q&A]
Hello,

I'm using a ClientResponseFilter to handle my rest client logging needs, it works great except when there's an i/o exception, in this case, a ProcessingException is thrown and handled by the whoever is invoking the rest call. so I have 2 questions :
1. Is there a way to handle ProcessingException in a centralized fashion just like what's used for ResponseExceptionMapper ?
2. Is there a filter/interceptor that could be implemented to intercept client i/o exceptions since ClientResponseFilter is not invoked in these cases ?

Thanks

[a comment by an automated account is not shown]

**Comment by jaiprasad266** [comment]
you can try handling the exception using ServerExceptionMapper
Follow the reference : https://quarkus.io/guides/resteasy-reactive#exception-mapping

Below is an example:

    @ServerExceptionMapper
    public Uni<RestResponse<String>> handleTimeoutException(IOException e) {
        return Uni.createFrom().item(RestResponse.status(Response.Status.INTERNAL_SERVER_ERROR,"Exception occurred"));
    }
